src/main/java/seedu/duke/parsermanager/ParseAddClient.java

```diff
@@ -52,42 +54,40 @@ public ParseAddClient(String addCommandDescription, ClientList clientList) {
         this.clientList = clientList;
     }
 
-    public Command parseCommand() throws EmptyDetailException, MissingFlagException, IncorrectFlagOrderException,
-            InvalidContactNumberException, InvalidEmailException, InvalidBudgetFormatException,
-            DuplicateClientException {
-        try {
-            checkForEmptyDetails(commandDescription);
-            ArrayList<String> clientDetails = processCommandAddClientDetails(commandDescription);
-            validateClientDetails(clientDetails);
-            return new CommandAddClient(clientDetails);
-        } catch (EmptyDetailException e) {
-            throw new EmptyDetailException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
-        } catch (MissingFlagException e) {
-            throw new MissingFlagException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
-        } catch (IncorrectFlagOrderException e) {
-            throw new IncorrectFlagOrderException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
+    public Command parseCommand() throws ParseAddClientException {
```

Comment:
I like how you used substitutability to avoid making the throws declaration too long